### DISCUSSION TITLE
importXML return an error on error popup instead of stuck on error popup

### DISCUSSION THREAD

**Discussion by DangerD1024** [CommandPost/CommandPost#3405] [Technical Support]
Currently on importXML when error popup appears (Wrong path, DTD validation failed, some other import error) script just pauses and waits for user interaction to click okay in popup.

Can importXML just return false if error popup appears instead of hanging?

**Comment by latenitefilms** [comment]
All `cp.apple.finalcutpro:importXML(path)` does is trigger some AppleScript:

```lua
--- cp.apple.finalcutpro:importXML(path) -> boolean
--- Method
--- Imports an XML file into Final Cut Pro
---
--- Parameters:
---  * path = Path to XML File
---
--- Returns:
---  * A boolean value indicating whether the AppleScript succeeded or not
function fcp:importXML(path)
    local appName = self:mainMenuName()
    local appleScript = [[
        set whichSharedXMLPath to "]] .. path .. [["
        tell application "]] .. appName .. [["
            activate
            open POSIX file whichSharedXMLPath as string
        end tell
    ]]
    local bool, _, _ = osascript.applescript(appleScript)
    return bool
end
```

It returns a boolean based on whether or not the AppleScript is successful or not.

Untested, but you MIGHT be able to use `cp.apple.finalcutpro:alert():isShowing()` to detect if an error message is showing?

**Reply by DangerD1024** [reply]
No, this will not work because whole commandpost stucks on this line:
`local bool, _, _ = osascript.applescript(appleScript)`
I've tried to run it in separate thread:
```
    local task = hs.task.new(
        "/usr/bin/osascript",
        function(exitCode, stdOut, stdErr)
            if exitCode ~= 0 then
                print("AppleScript execution failed: " .. (stdErr or "Unknown error"))
            else
                print("AppleScript executed successfully.")
            end
        end,
        { "-e", appleScript }
    )
```
But it's not starting by some reason...

**Reply by latenitefilms** [reply]
Huh, interesting! The AppleScript call itself must be blocking.
